jobs/gtfs-rt-parser/gtfs_rt_parser.py

```diff
@@ -123,109 +218,440 @@ def identify_files(glob, rt_file_type: RTFileType, progress=False) -> List[RTFil
+                        f"WARNING: no records found in {str(rt_file.path)}",
+                        fg=typer.colors.YELLOW,
+                        pbar=pbar,
+                    )
+                continue
+
             for record in parsed["entity"]:
-                record.update(
-                    {
-                        "header": parsed["header"],
-                        "metadata": json.loads(
-                            rt_file.json()
-                        ),  # back and forth so we use pydantic serialization
-                    }
+                gzipfile.write(
+                    (
+                        json.dumps(
+                            {
+                                "header": parsed["header"],
+                                # back and forth so we use pydantic serialization
+                                "metadata": json.loads(rt_file.json()),
+                                **copy.deepcopy(record),
+                            }
+                        )
+                        + "\n"
+                    ).encode("utf-8")
                 )
-                gzipfile.write((json.dumps(record) + "\n").encode("utf-8"))
                 written += 1
+            del parsed
 
-        out_path = f"{rt_file.hive_path(bucket)}{EXTENSION}"  # probably a better way to do this
-        msg = f"writing {written} lines from {str(rt_file.path)} to {out_path}"
-        if pbar:
-            pbar.write(msg)
-        else:
-            typer.echo(msg)
-        put_with_retry(fs, gzip_fname, out_path)
-        if pbar:
-            pbar.update(1)
+    if written:
+        log(
+            f"writing {written} lines to {hour.data_hive_path}{JSONL_GZIP_EXTENSION}",
+            pbar=pbar,
+        )
+        put_with_retry(fs, gzip_fname, f"{hour.data_hive_path}{JSONL_GZIP_EXTENSION}")
+    else:
+        log(
+            f"WARNING: no records at all for {hour.data_hive_path}",
+            fg=typer.colors.YELLOW,
+            pbar=pbar,
+        )
 
+    if errors:
+        upload_gzipped_records(
+            fs,
+            gzip_fname=str(tmp_dir + f"/errors_{hour.suffix}" + JSONL_GZIP_EXTENSION),
+            out_path=f"{hour.errors_hive_path}{JSONL_GZIP_EXTENSION}",
+            records=errors,
+            pbar=pbar,
+            verbose=verbose,
+        )
+    else:
+        log(
+            f"WARNING: no records at all for {hour.errors_hive_path}",
+            fg=typer.colors.YELLOW,
+            pbar=pbar,
+        )
 
-def try_handle_one_feed(*args, **kwargs) -> Optional[Exception]:
-    try:
-        parse_file(*args, **kwargs)
-    except Exception as e:
-        typer.echo(
-            f"got exception while handling feed: {str(e)} from {str(e.__cause__ or e.__context__)} {traceback.format_exc()}",
-            err=True,
+
+# Originally this whole function was retried, but tmpdir flakiness will throw
+# exceptions in backoff's context, which ruins things
+def parse_and_aggregate_hour(
```

Comment:
**question**: Does this need to have `hour` hardcoded as part of name? Version of same question above. And should this be renamed to be like `process` or something to account for both parsing and validation?